### ISSUE TITLE
Copy-pasting a valid link results in the content (span) of the link to be the page title instead of the URL

### ISSUE THREAD

**Issue by Illyism** [marktext/marktext#1309]
### Describe your feature request

When journaling or writing and referencing some article, it will be a lot easier to copy and paste links if they automatically form into the title of the page instead of the URL.

This way you don't have to give a proper name to the link right away and you can keep typing.

### Example

Current behaviour:

```markdown
Paste: https://en.wikipedia.org/wiki/Markdown
Result: [https://en.wikipedia.org/wiki/Markdown](https://en.wikipedia.org/wiki/Markdown)
```

Improved behaviour:

```markdown
Paste: https://en.wikipedia.org/wiki/Markdown
Result: [Markdown - Wikipedia](https://en.wikipedia.org/wiki/Markdown)
```

**Comment by Jocs** [comment]
This is a good feature, it will improve the user experience very much. as much as I see, the link need to be a html page.

PR is welcomed.

**Comment by fxha** [comment]
@Jocs Was this issue resolved in #1344? BTW this only work with Chromium based browsers and not Firefox. If I c&p a URL in Firefox it will result in a raw URL instead a markdown URL (`![]()`).

**Comment by Jocs** [comment]
>Was this issue resolved in #1344? BTW this only work with Chromium based browsers and not Firefox. If I c&p a URL in Firefox it will result in a raw URL instead a markdown URL (![]()).

I didn't test Firefox, I'll test it later
